esp/src/eclwatch/PreflightDetailsWidget.js

```diff
@@ -47,95 +47,186 @@ define([
                 this.openButton = registry.byId(this.id + "Open");
                 this.openButton.set("disabled", true);
 
-                var retVal = new declare([ESPUtil.Grid(true, true)])({
+                var retVal = new declare([ESPUtil.Grid(true, false, {rowsPerPage:1000}, true)])({
                     store: this.store,
-                    columns: context.setColumns()
+                    columns: this.setColumns()
                 }, domID);
                 return retVal;
             },
 
+            calculateColumnWidth: function (grid) {
+                for (var key in grid.columns) {
+                    console.log(grid.columns[key].id)
+                    if (grid.columns[key].label !== "Location" && grid.columns[key].label !== "Component") { //this needs improvement
+                        var node = domConstruct.toDom('<div style="position:absolute;visibility:hidden">' + grid.columns[key].label + '</div>');
+                        domConstruct.place(node, win.body());
+                        var p = domGeom.position(node);
+                        domConstruct.destroy(node);
```

Comment:
You might want to create it once outside the loop and delete it after the loop as finished?
(Then just update the innerHTML / innerText for each iteration.

Reply:
I only need it for measurement purposes and then I dispose of it quickly thereafter for each column.

Reply:
Will test.

Reply:
Yes - I am only suggesting your create the dom once and reuse it for each column (by setting its innerText) and then clean it up (quicker).